gluten-core/src/main/java/io/glutenproject/memory/MemoryUsageStats.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.memory;
+
+public class MemoryUsageStats {
+  public final long peak;
+  public final long current;
+
+  public MemoryUsageStats(MemoryUsage usage) {
```

Comment:
Could we pass just long value?

Reply:
The class should be tightly coupled with MemoryUsage.java. We can treat it as a snapshot to MemoryUsage instances.

I'll enclose the ctor to package-private.